docker/build.sh

```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+
+# Build the `test-server` image, which runs the server in a simple configuration
+# designed to be used in tests, based on the current git revision.
+
+TAG="${1:-latest}"
+ETESYNC_VERSION=$(git describe --tags HEAD)
+git archive HEAD -o docker/test-server/archive.tar
```

Comment:
No need for this, the COPY in Dockerfile already does this (just make sure to have a .dockerignore file).
The dockerignore file could look something like this:
```
/db.sqlite3*
Session.vim
/local_settings.py
.venv
/assets
/logs
/.coverage
/tmp
/media
.git
.vim

__pycache__
.*.swp

/docker
```

Reply:
That can still end up copying lots of extra stuff, possibly including secrets or other large temporary files that happen to be in this directory.  I've been burned by that before!  The archive has the advantage of building exactly what `git describe` says it builds, in a more-or-less reproducible way.  If there was a `.dockerinclude` file things might be different, but and exclusion list is always going to be problematic.

Reply:
I realise that, and I also dislike it, but that's standard Docker stuff. Doing something different than what's standard is a cause for confusion and more trouble down the line. The proper place to fix this is docker.

Reply:
OK, fair enough!